### PR TITLE
Clear log

### PR DESCRIPTION
**Pull request by UziTech**
### Description of the Change

This adds a clear notifications button to the log

image

### Alternate Designs

At the moment the button clears the notifications with `atom.notifications.clear()` as well as clearing the log and any displayed notifications.

I would like to get it working with [`atom.notifications.onDidClearNotifications()`](https://github.com/atom/atom/pull/16074) once that gets released.

### Benefits

Ability to clear the notifications log

### Possible Drawbacks

none

### Applicable Issues

https://github.com/atom/notifications/pull/164#issuecomment-341341995

### CODE REVIEW

**Comment by 50Wliu** [comment]
Hmm.  I can't say I'm the biggest fan of the dash meaning "clear", but I also don't see any better octicon.  Have you considered maybe just a button that says "Clear All" (or some variant)?

**Comment by UziTech** [comment]
I did have the button say clear initially. The problem I saw was when the Logs dock is on the right the button took up too much real estate for how important it is. It would be nice to find an icon that would be intuitive.

**Comment by 50Wliu** [comment]
@simurai do you have any thoughts about this?  On my screen there's definitely enough space for text when in a side dock.
notifications-clear-all-space

Maybe the Clear button could wrap to a new line if there isn't enough room so that we can keep it as text.

**Comment by simurai** [comment]
Maybe `trashcan`?

image

Like "deleting" all notifications.

**Comment by simurai** [comment]
Or `circle-slash`?

screen shot 2018-01-03 at 3 09 40 pm

DevTools uses that to clear the console. Although we have been using it for the "forbidden/not allowed" cursor.

**Comment by UziTech** [comment]
I like the trashcan. That seems to be the most intuitive.

File: lib/main.coffee [hunk 1]
```diff
@@ -61,6 +61,13 @@ Notifications =
     @addNotificationsLogSubscriptions() if @notificationsLog?
     @subscriptions.add atom.workspace.addOpener (uri) => @createLog() if uri is NotificationsLog::getURI()
     @subscriptions.add atom.commands.add 'atom-workspace', 'notifications:toggle-log', -> atom.workspace.toggle(NotificationsLog::getURI())
+    @subscriptions.add atom.commands.add 'atom-workspace', 'notifications:clear-log', =>
+      for notification in atom.notifications.getNotifications()
+        notification.options.dismissable = true
+        notification.dismissed = false
```
Comment (50Wliu) [inline lib/main.coffee]: Why is this needed?  Under what circumstances will there be an already-dismissed notification in the list that we need to dismiss again?

Comment (UziTech) [reply lib/main.coffee]: non-dimissable notifications (ones that dismiss automatically) have [`.dismissed` = true](https://github.com/atom/atom/blob/a3108623de406e6f162c37b2dffd0872ab39e276/src/notification.js#L13) when they are first shown. If someone clears the log we want those notifications to be dismissed right away not wait for the timer to dismiss them automatically.

**Comment by UziTech** [comment]
Can this get reviewed?

File: styles/notifications-log.less [hunk 2]
```diff
@@ -24,6 +24,10 @@
       opacity: .5;
       background: @base-background-color;
     }
+
+    .notifications-clear-log {
+      float: right;
+    }
```
Comment (jasonrudolph) [inline styles/notifications-log.less]: @UziTech: When you have a moment, can you share screenshots of this new icon is rendered in each of the UI themes that ship with Atom? I suspect it renders just fine, but we want to make sure. 😅

Comment (UziTech) [reply styles/notifications-log.less]: ### Atom Dark
image

### Atom Light
image

### One Dark
image

### One Light
image

**Comment by UziTech** [comment]
I rebased and fixed the issues

**Comment by jasonrudolph** [comment]
> I rebased and fixed the issues

Cool! Thanks for your perseverance on this pull request, @UziTech. :zap:
